app/src/main/java/com/google/android/ground/ui/map/gms/mog/MogTileDownloader.kt

```diff
@@ -33,17 +36,20 @@ class MogTileDownloader(private val client: MogClient, private val outputBasePat
    * Executes the provided [requests], writing resulting tiles to [outputBasePath] in sub-paths of
    * the form `{z}/{x}/{y}.jpg`.
    */
-  suspend fun downloadTiles(requests: List<MogTilesRequest>) = flow {
-    client.getTiles(requests).collect { tile ->
-      val outFile = File(outputBasePath, tile.metadata.tileCoordinates.getTilePath())
-      outFile.parentFile?.mkdirs()
-      val gmsTile = tile.toGmsTile()
-      val data = gmsTile.data!!
-      outFile.writeBytes(data)
-      Timber.d("Saved ${data.size} bytes to ${outFile.path}")
-      emit(data.size)
-    }
-  }
+  suspend fun downloadTiles(requests: List<MogTilesRequest>) =
+    flow {
+        client.getTiles(requests).collect { tile ->
+          currentCoroutineContext().ensureActive()
```

Comment:
Instead of manually checking for this, we can make the flow returned by `getTiles` in `MogClient` cancelable. 

[documentation](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.flow/cancellable.html#:~:text=Returns%20a%20flow%20which%20checks,operator%20provides%20a%20shortcut%20for%20.)

Reply:
just formatting in this pr, not a part of this PR

Reply:
This was added in this PR itself. Not a formatting. Please check again.